test/unix/test_lwt_unix.cppo.ml

```diff
@@ -726,6 +726,47 @@ let lwt_preemptive_tests = [
   end;
 ]
 
+let lwt_user_tests = [
+  test "getlogin and Unix.getlogin" ~only_if:(fun () -> not Sys.win32) begin fun () ->
+    let unix_user = Unix.getlogin () in
+    Lwt_unix.getlogin () >>= fun user ->
+    Lwt.return (user = unix_user)
+  end;
+  test "getpwnam and Unix.getpwnam" ~only_if:(fun () -> not Sys.win32) begin fun () ->
+    let unix_user = Unix.getlogin () in
+    let unix_password = Unix.getpwnam unix_user in
+    Lwt_unix.getpwnam unix_user >>= fun password ->
+    Lwt.return (password = unix_password)
+  end;
+  test "getpwuid and Unix.getpwuid" ~only_if:(fun () -> not Sys.win32) begin fun () ->
+    let pwnam = Unix.getpwnam (Unix.getlogin ()) in
+    let unix_pwuid = Unix.getpwuid pwnam.pw_uid in
+    Lwt_unix.getpwuid pwnam.pw_uid >>= fun pwuid ->
+    Lwt.return (pwuid = unix_pwuid)
+  end;
+  test "getgrgid and Unix.getgrgid" ~only_if:(fun () -> not Sys.win32) begin fun () ->
+    let groups = Unix.getgroups () in
```

Comment:
Have you considered `Unix.getgid` instead? It looks like it should always return a gid for the test to use, so you don't have to have the pattern-matching below, and a fall-through case.

Reply:
Updated to use `Unix.getgid` instead.